docs/design/vcpkg.md

```diff
@@ -0,0 +1,159 @@
+# Adding Cargo support to Hermeto
+
+## Background
+
+[vcpkg][] is a C/C++ package manager maintained by Microsoft. At the moment of
+writing it is not abandoned and supports quite a few (about 2600)  C++
+libraries. Support for this PM was inquired about by users.  vcpkg supports
+multiple build systems, provides both default registry as well as allows
+setting up custom ones, provides binaries and sources, [the documentation][]
+specifically mentions air-gapped environments which in theory maps nicely to
+hermetic builds concept. Despite being advertised the underlying feature
+necessary for hermetic builds is [considered experimental][] at the moment of
```

Comment:
>  The X_VCPKG_ASSET_SOURCES environment variable, or
>  The --x-asset-sources command-line option.

Yeah, which means our implementation would have to remain experimental as well until ^this is resolved upstream, because if they dropped it, we'd have a broken, yet supported, implementation.